pkg/agent/route/route_linux.go

```diff
@@ -1478,6 +1532,62 @@ func (c *Client) DeleteLocalAntreaFlexibleIPAMPodRule(podAddresses []net.IP) err
 	return nil
 }
 
+// addNodeIP adds nodeIP into the ipset when a new Node joins the cluster.
+// The ipset is consumed with encap mode when multicast is enabled.
+func (c *Client) addNodeIP(podCIDR *net.IPNet, nodeIP net.IP) error {
+	if !c.multicastEnabled || !c.networkConfig.TrafficEncapMode.SupportsEncap() {
+		return nil
+	}
+	if nodeIP == nil {
+		return nil
+	}
+	ipSetEntry := nodeIP.String()
+	if nodeIP.To4() != nil {
+		if err := ipset.AddEntry(clusterNodeIPSet, ipSetEntry); err != nil {
+			return err
+		}
+		c.clusterNodeIPs.Store(podCIDR.String(), ipSetEntry)
+	} else {
+		if err := ipset.AddEntry(clusterNodeIP6Set, ipSetEntry); err != nil {
+			return err
+		}
+		c.clusterNodeIP6s.Store(podCIDR.String(), ipSetEntry)
+	}
+	return nil
+}
+
+// deleteNodeIP deletes NodeIPs from the ipset when a Node leaves the cluster.
+// The ipset is consumed with encap mode when multicast is enabled.
+func (c *Client) deleteNodeIP(podCIDR *net.IPNet) error {
+	if !c.multicastEnabled || !c.networkConfig.TrafficEncapMode.SupportsEncap() {
+		return nil
+	}
+
+	podCIDRStr := podCIDR.String()
+	if podCIDR.IP.To4() != nil {
+		obj, exists := c.clusterNodeIPs.Load(podCIDRStr)
+		ipSetEntry := obj.(string)
+		if !exists {
+			return nil
+		}
+		if err := ipset.DelEntry(clusterNodeIPSet, ipSetEntry); err != nil {
+			return err
+		}
+		c.clusterNodeIPs.Delete(ipSetEntry)
```

Comment:
This uses wrong key, nothing will be deleted.
There are few mistakes in this file, which makes me feel an unit test is required, especially that we don't test this in IPv6 testbed.